### DISCUSSION TITLE
v1.14.0

### DISCUSSION THREAD

**Discussion by kyleconroy** [sqlc-dev/sqlc#1671] [Announcements]
## What's Changed
* fix(bundler): Only close multipart writer once by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1528
* Adding LastInsertId for Go by @Xeoncross in https://github.com/kyleconroy/sqlc/pull/1484
* build(deps): bump github.com/lib/pq from 1.10.4 to 1.10.5 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1534
* build(deps): bump actions/setup-java from 2 to 3 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1542
* build(deps): bump actions/setup-go from 2 to 3 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1543
* build(deps): bump golang from 1.18.0 to 1.18.1 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1547
* Added update query to postgresql docs by @airscholar in https://github.com/kyleconroy/sqlc/pull/1535
* feat: run sqlc with docker on windows cmd by @ParvinEyvazov in https://github.com/kyleconroy/sqlc/pull/1557
* Pydantic models config field by @danicc097 in https://github.com/kyleconroy/sqlc/pull/1538
* build(deps): bump github.com/jackc/pgx/v4 from 4.15.0 to 4.16.0 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1562
* Prepend an underscore to a StructName if it starts with a digit. by @avalonbits in https://github.com/kyleconroy/sqlc/pull/1554
* fix typo in feature request bug template by @josharian in https://github.com/kyleconroy/sqlc/pull/1568
* feat: Add JSON "codegen" output by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1565
* build(deps): bump github.com/google/go-cmp from 0.5.7 to 0.5.8 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1573
* UPDATED: sqlc.narg() - allow user-specified nullability by @skabbes in https://github.com/kyleconroy/sqlc/pull/1536
* docs: add documentation for sqlc.narg() by @skabbes in https://github.com/kyleconroy/sqlc/pull/1580
* docs: Add some more example code for running transactions by @Jille in https://github.com/kyleconroy/sqlc/pull/1583
* build(deps): bump github.com/jackc/pgx/v4 from 4.16.0 to 4.16.1 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1597
* Add 1.13.0 to Issue Template by @danielbprice in https://github.com/kyleconroy/sqlc/pull/1592
* Add support for CHANGE COLUMN in MySQL by @dansimau in https://github.com/kyleconroy/sqlc/pull/1605
* build(deps): bump golang from 1.18.1 to 1.18.2 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1609
* docs: fix examples by @orisano in https://github.com/kyleconroy/sqlc/pull/1608
* Fix panic when calling cmd.Do subsequently by @jlisthood in https://github.com/kyleconroy/sqlc/pull/1614
* build(deps): bump github.com/lib/pq from 1.10.5 to 1.10.6 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1627
* doc: typo fix in overrides config example by @reddec in https://github.com/kyleconroy/sqlc/pull/1619
* [FIX] copyfrom imports by @positiveblue in https://github.com/kyleconroy/sqlc/pull/1626
* internal/codegen: add Enum.Valid and AllEnumValues by @josharian in https://github.com/kyleconroy/sqlc/pull/1613
* Refactor to Package Catalog to Improve Readability by @jakoguta in https://github.com/kyleconroy/sqlc/pull/1595
* Validate sqlc function arguments by @ryan-berger in https://github.com/kyleconroy/sqlc/pull/1633
* Bug fix to index relation creation bug by @jakoguta in https://github.com/kyleconroy/sqlc/pull/1640
* docs: Add documentation for version 2 of the configuration file by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1657
* build(deps): bump golang from 1.18.2 to 1.18.3 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1658
* docs: Re-org the config page. Add JSON docs by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1659
* build(deps): bump actions/setup-python from 3 to 4 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1666
* add support for custom Go struct tags by @josharian in https://github.com/kyleconroy/sqlc/pull/1569
* fix: fixed typo `sql.narg` in doc by @akhilmhdh in https://github.com/kyleconroy/sqlc/pull/1668
* feat: Process-based codegen plugins by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1578
* cmd/sqlc: Bump version to v1.14.0 by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1670

## New Contributors
* @Xeoncross made their first contribution in https://github.com/kyleconroy/sqlc/pull/1484
* @airscholar made their first contribution in https://github.com/kyleconroy/sqlc/pull/1535
* @ParvinEyvazov made their first contribution in https://github.com/kyleconroy/sqlc/pull/1557
* @danicc097 made their first contribution in https://github.com/kyleconroy/sqlc/pull/1538
* @avalonbits made their first contribution in https://github.com/kyleconroy/sqlc/pull/1554
* @josharian made their first contribution in https://github.com/kyleconroy/sqlc/pull/1568
* @skabbes made their first contribution in https://github.com/kyleconroy/sqlc/pull/1536
* @danielbprice made their first contribution in https://github.com/kyleconroy/sqlc/pull/1592
* @jlisthood made their first contribution in https://github.com/kyleconroy/sqlc/pull/1614
* @reddec made their first contribution in https://github.com/kyleconroy/sqlc/pull/1619
* @positiveblue made their first contribution in https://github.com/kyleconroy/sqlc/pull/1626
* @jakoguta made their first contribution in https://github.com/kyleconroy/sqlc/pull/1595
* @ryan-berger made their first contribution in https://github.com/kyleconroy/sqlc/pull/1633
* @akhilmhdh made their first contribution in https://github.com/kyleconroy/sqlc/pull/1668

**Full Changelog**: https://github.com/kyleconroy/sqlc/compare/v1.13.0...v1.14.0

<hr /><em>This discussion was created from the release <a href='https://github.com/kyleconroy/sqlc/releases/tag/v1.14.0'>v1.13.0</a>.</em>

**Comment by roccoblues** [comment]
Thanks for the new release! I guess this announcement should be called 1.14?

Is it intended that the new version requires at least Go 1.18? Looks like the code could be easily changed to work without generics.

```
github.com/kyleconroy/sqlc@v1.14.0/internal/config/go_type.go:175:15: undefined: strings.Cut
note: module requires Go 1.18
```

**Comment by kyleconroy** [comment]
> I guess this announcement should be called 1.14?

Fixed

> Is it intended that the new version requires at least Go 1.18

Yep. While the code that requires it is trivial to fix, 1.18 has been out for three months. I'm sure we're going to have more place where we want to use newer features in the future, so I'm happy with requiring 1.18 now.

The generated code does not require require 1.18
